### DISCUSSION TITLE
[Bug]: netclient version 17.1 for rasberry pi 4 version

### DISCUSSION THREAD

**Discussion by bose2273** [gravitl/netmaker#2288] [Q&A] [answered]
### Contact Details

surajit2273@gmail.com

### What happened?

Can any body help me pls!!

I want to install the netclient 17.1 in my raspberry pi 4 Model B. Which netclient should i use? I tried the "netclient-arm7" didn't work. got error. Any suggestion or help will be greatly appreciated.

### Version

v0.17.1

### What OS are you using?

Linux

### Relevant log output

```shell
root@raspberrypi:/# sudo dpkg -i netclient-arm7
dpkg-deb: error: 'netclient-arm7' is not a Debian format archive
dpkg: error processing archive netclient-arm7 (--install):
dpkg-deb --control subprocess returned error exit status 2
Errors were encountered while processing:
netclient-arm7
```

### Contributing guidelines

- [X] Yes, I did.

**Comment by mattkasun** [comment, marked as answer]
the architecture of Pi4 is arm64

**Reply by bose2273** [reply]
Thank you so much!!

**Reply by bose2273** [reply]
HI Mattkasun

I have installed the "netclient_0.17.1_arm64.deb" as i am running ubuntu but getting the below error

[netclient] 2023-05-11 18:53:55 joining labnet at api.netmaker.telcasta.com:443 
[netclient] 2023-05-11 18:53:55 network: labnet UDP hole punching enabled for node ubuntu 
[netclient] 2023-05-11 18:53:55 starting wireguard 
[netclient] 2023-05-11 18:53:59 error running command: systemctl restart netclient.service 
[netclient] 2023-05-11 18:53:59 Failed to restart netclient.service: Unit netclient.service not found. 

and when i do systemctl status i get root@ubuntu:~# systemctl status netclient
Unit netclient.service could not be found.

any suggestion?

Thanks in adv!!

**Reply by bose2273** [reply]
Below is the specs 

cat /proc/cpuinfo
processor : 0
model name : ARMv7 Processor rev 3 (v7l)
BogoMIPS : 108.00
Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x0
CPU part : 0xd08
CPU revision : 3

processor : 1
model name : ARMv7 Processor rev 3 (v7l)
BogoMIPS : 108.00
Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x0
CPU part : 0xd08
CPU revision : 3

processor : 2
model name : ARMv7 Processor rev 3 (v7l)
BogoMIPS : 108.00
Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x0
CPU part : 0xd08
CPU revision : 3

processor : 3
model name : ARMv7 Processor rev 3 (v7l)
BogoMIPS : 108.00
Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x0
CPU part : 0xd08
CPU revision : 3

Hardware : BCM2711
Revision : b03112
Serial : 10000000987bf8f6
Model : Raspberry Pi 4 Model B Rev 1.2

total        used        free      shared  buff/cache   available
Mem:          1.8Gi       191Mi       1.1Gi       2.0Mi       482Mi       1.5Gi
Swap:            0B          0B          0B

**Reply by mattkasun** [reply]
I will have to look into the arm64 Deb.  

As workaround downloaded the netclient-linux-arm64 from the releases on GitHub.  rename the downloaded file to netmaker, make it executable, and run
`./netclient install`

**Comment by bose2273** [comment]
Worked liked a charm!!! Thank you so much ..

Just one thing there is no release name "netclient-linux-arm64" .. its netclient-arm64 

Thanks once again for everything.

Regards
